ui/imports/shared/popups/send/Helpers.qml

```diff
@@ -0,0 +1,69 @@
+pragma Singleton
+
+import QtQuick 2.15
+import QtQml 2.15
+
+import utils 1.0
+import shared.stores 1.0
+import shared.stores.send 1.0
+import AppLayouts.Wallet.stores 1.0 as WalletStores
+
+import StatusQ.Core 0.1
+
+import "./panels"
+import "./controls"
+import "./views"
+
+QtObject {
+    id: root
+
+    function createSendModalRequirements() {
+        return {
+            preSelectedAccount: null,
+            preSelectedRecipientType: TabAddressSelectorView.Type.Address,
+            preSelectedRecipient: null,
+            preSelectedHoldingType: 0,
+            preSelectedHolding: null,
+            preSelectedHoldingID: "",
+            preDefinedAmountToSend: "",
+            preSelectedSendType: Constants.SendType.Transfer
+        }
+    }
+
+    function lookupAddressesForSendModal(senderAddress, recipientAddress, asset, isCollectible, amount) {
+        let req = createSendModalRequirements()
+
+        req.preSelectedSendType = Constants.SendType.Transfer
+        let senderAccount = null
+        let resolvedAcc = WalletStores.RootStore.lookupAddressObject(senderAddress)
+        if (resolvedAcc && resolvedAcc.type == WalletStores.RootStore.LookupType.Account) {
+            req.preSelectedAccount = resolvedAcc.object
+        }
+
+        let res = WalletStores.RootStore.lookupAddressObject(recipientAddress)
+        if (res) {
+            if (res.type == WalletStores.RootStore.LookupType.Account) {
+                req.preSelectedRecipientType = TabAddressSelectorView.Type.Account
+                req.preSelectedRecipient = res.object
+            } else if (res.type == WalletStores.RootStore.LookupType.SavedAddress) {
+                req.preSelectedRecipientType = TabAddressSelectorView.Type.SavedAddress
+                req.preSelectedRecipient = res.object
+            }
+        } else {
+            req.preSelectedRecipientType = TabAddressSelectorView.Type.Address
+            req.preSelectedRecipient = recipientAddress
+        }
+
+        if (isCollectible) {
+            req.preSelectedHoldingType = Constants.HoldingType.Collectible
+            req.preSelectedHolding = asset
```

Comment:
Is the expected `asset` the complete token `object` or and `id`? I think you need to review `lines 59` and/or `62`!!

Reply:
The asset is either a complete object representing an NFT or a symbol representing ERC20 or native currency.

Reply:
The comments were lost in the last refactoring, will update the function description to clarify this.